apps/demos/Demos/Chat/AIIntagration/jQuery/index.js

```diff
@@ -0,0 +1,189 @@
+const deployment = 'gpt-4o-mini';
+const apiVersion = '2024-02-01';
+const endpoint = 'https://public-api.devexpress.com/demo-openai';
+const apiKey = 'DEMO';
+
+const store = [];
+let messages = [];
+let lastAIResponse = null;
+
+const user = {
+  id: 'user',
+};
+
+const assistant = {
+  id: 'assistant',
+  name: 'Virtual Assistant',
+};
+
+$(() => {
+  DevExpress.localization.loadMessages({
+    "en": {
+      "dxChat-emptyListMessage": "Virtual Assistant",
+      "dxChat-emptyListPrompt": "An AI bot ready to provide instant support for your needs.",
+    },
+  });
+
+  const chatService = new AzureOpenAI({
+    dangerouslyAllowBrowser: true,
+    deployment,
+    endpoint,
+    apiVersion,
+    apiKey,
+  });
+
+  async function getAIResponse(messages) {
+    const params = {
+      messages: messages,
+      max_tokens: 1000,
+      temperature: 0.7,
+    };
+
+    const responseAzure = await chatService.chat.completions.create(params);
+    const data = { choices: responseAzure.choices };
+
+    return data.choices[0].message?.content;
+  };
+
+  async function processMessageSending() {
+    instance.option({ typingUsers: [assistant] });
+
+    try {
+      const aiResponse = await getAIResponse(messages);
+
+      setTimeout(() => {
+        instance.option({ typingUsers: [] });
+
+        messages.push({ role: 'assistant', content: aiResponse });
+
+        renderMessage(aiResponse);
+      }, 200);
+    } catch {
+      instance.option({
+        typingUsers: [],
+        alerts: [{ message: 'Request limit reached, try again in a minute.' }],
+      });
+
+      if (lastAIResponse) {
+        messages.push(lastAIResponse);
+        renderMessage(lastAIResponse.content);
+        lastAIResponse = null;
+      }
```

Comment:
Do I understand correctly that it's for a case if we try regenerate but tokens limit is reached? I think it's okay to render nothing in this case but simplify the code

Reply:
The “catch the error earlier” approach is not applicable here. Since we need to show the user that the regeneration process is in progress.

To get rid of the cache, I used a different approach: I refresh the last message. I think this is a good way to show a more linear code and also the CustomStore features.

Reply:
[84432256f04f4b45bd414c7b2a08bf412a9421bd]